cmd/agent/subcommands/run/command.go

```diff
@@ -196,69 +196,48 @@ func run(log log.Component, config config.Component, cliParams *cliParams) error
 
 // StartAgentWithDefaults is a temporary way for other packages to use startAgent.
 func StartAgentWithDefaults() error {
-	return startAgent(&cliParams{GlobalParams: &command.GlobalParams{}})
+	// run startAgent in an app, so that the log and config components get initialized
+	return fxutil.OneShot(func(log log.Component, config config.Component) error {
+		return startAgent(&cliParams{GlobalParams: &command.GlobalParams{}})
+	},
+		fx.Supply(core.BundleParams{
+			ConfFilePath:      "", // no config file path specification in this situation
```

Comment:
I think this is maintaining the existing behavior, but it seems a bit strange to me that we don't allow configuration of the config file paths when running as a service. Something to think about for the future.

Reply:
That's my understanding as well.

Reply:
There are some inconsistencies, for example, see https://github.com/DataDog/datadog-agent/pull/13842#discussion_r1001605760

Reply:
@ogaca-dd are you suggesting that this part of the diff is incorrect, or just linking for more context on inconsistent configuration file paths?